vernac/vernacentries.ml

```diff
@@ -1856,6 +1856,13 @@ let subproof_kind = Proof.new_focus_kind ()
 let subproof_cond = Proof.done_cond subproof_kind
 
 let vernac_subproof gln =
+  let gln =
+    match gln with
+    | None -> None
+    | Some (SelectNth n) -> Some n
+    | _ -> user_err ~hdr:"bracket_selector"
+             (str "Brackets only support the single numbered goal selector.")
+  in
   Proof_global.simple_with_current_proof (fun _ p ->
     match gln with
     | None -> Proof.focus subproof_cond () 1 p
```

Comment:
Why not directly change this match instead of the above let-in?
Also we could expose `Proofs._focus` (with a different name ofc) to allow focusing on a range of goals.

Reply:
> Why not directly change this match instead of the above let-in?

Will update. This was the result of first using your function `query_command_selector` and then inlining its code with a different error message. I didn't realize that further simplification was possible